### PR TITLE
Feature: Move open orders to func component

### PR DESCRIPTION
**Pull request by VladyslavP**

### CODE REVIEW

**Review by akhlopiachyi — APPROVED** [review]

File: src/containers/OpenOrders/index.tsx [hunk 1]
```diff
@@ -29,152 +26,120 @@ interface ReduxProps {
...
-}
 
-const mapStateToProps = (state: RootState): ReduxProps => ({
-    currentMarket: selectCurrentMarket(state),
-    list: selectOpenOrdersList(state),
-    fetching: selectOpenOrdersFetching(state),
-    cancelFetching: selectCancelOpenOrdersFetching(state),
-    userLoggedIn: selectUserLoggedIn(state),
-});
+    const handleCancelAll = React.useCallback(() => {
```
Comment (oyershov) [inline src/containers/OpenOrders/index.tsx:114]: use, hook here

File: src/containers/OpenOrders/index.tsx [hunk 2]
```diff
@@ -29,152 +26,120 @@ interface ReduxProps {
...
-    openOrdersCancelFetch: payload => dispatch(openOrdersCancelFetch(payload)),
-    ordersCancelAll: payload => dispatch(ordersCancelAllFetch(payload)),
-});
+    const classNames = React.useMemo(() => classnames('pg-open-orders', {
+        'pg-open-orders--empty': !list.length,
+        'pg-open-orders--loading': fetching,
+    }), [list.length, fetching]);
 
-export type OpenOrdersProps = ReduxProps;
+    React.useEffect(() => {
```
Comment (oyershov) [inline src/containers/OpenOrders/index.tsx:123]: use hook here

**Review by oyershov — CHANGES_REQUESTED** [review]

File: src/containers/OpenOrders/index.tsx [hunk 3]
```diff
@@ -29,152 +26,104 @@ interface ReduxProps {
...
+            formatMessage({ id: 'page.body.trade.header.openOrders.content.total' }).concat(currentBidUnit),
+            formatMessage({ id: 'page.body.trade.header.openOrders.content.filled' }),
+            '',
+        ];
+    }, [base_unit, quote_unit, formatMessage]);
 
-    private handleCancelAll = () => {
-        const { currentMarket } = this.props;
-        currentMarket && this.props.ordersCancelAll({ market: currentMarket.id });
+    const openOrders = () => {
```
Comment (oyershov) [inline src/containers/OpenOrders/index.tsx:91]:
```suggestion
    const renderOpenOrders = () => {
```

File: src/hooks/useOpenOrdersFetch.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { userOpenOrdersFetch } from '../modules/user/openOrders';
+import { selectUserLoggedIn } from '../modules/user/profile';
+
+export const useOpenOrdersFetch = ({ id }) => {
```
Comment (oyershov) [inline src/hooks/useOpenOrdersFetch.ts:6]: Pass whole market object here
```suggestion
export const useOpenOrdersFetch = ({ market: Market }) => {
```

File: src/hooks/useOpenOrdersFetch.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    const dispatch = useDispatch();
+    const userLoggedIn = useSelector(selectUserLoggedIn);
+
+    React.useEffect(() => {
+        if (userLoggedIn && id) {
+            dispatch(userOpenOrdersFetch({ market: { id }} as any));
+        }
+    }, [userLoggedIn, id, dispatch]);
```
Comment (oyershov) [inline src/hooks/useOpenOrdersFetch.ts:14]:
```suggestion
    }, [userLoggedIn, market.id, dispatch]);
```

**Review by oyershov — CHANGES_REQUESTED** [review]

File: src/hooks/useOpenOrdersFetch.ts (hunk 5 above)
Comment (berezh) [reply src/hooks/useOpenOrdersFetch.ts:14]: no need to pass `dispatch` and other static methods or vars

File: src/containers/OpenOrders/index.tsx [hunk 6]
```diff
@@ -29,152 +26,104 @@ interface ReduxProps {
...
-        );
+    const handleCancel = (index: number) => {
+        const orderToDelete = list[index];
+        dispatch(openOrdersCancelFetch({ order: orderToDelete, list }));
     };
 
-    private renderData = () => {
-        const { list, currentMarket } = this.props;
-
+    const renderData = () => {
```
Comment (berezh) [inline src/containers/OpenOrders/index.tsx:52]: should be `useCallback` method with dependecies

File: src/containers/OpenOrders/index.tsx [hunk 7]
```diff
@@ -1,180 +1,122 @@
...
-) as React.FunctionComponent;
+    return (
+        <div className={classNames}>
+            <div className="cr-table-header__content">
+                <div className="cr-title-component">
+                    <FormattedMessage id="page.body.trade.header.openOrders" />
+                    <span className="cr-table-header__cancel" onClick={handleCancelAll}>
+                            <FormattedMessage id="page.body.openOrders.header.button.cancelAll" />
+                            <CloseIcon className="cr-table-header__close" />
+                        </span>
```
Comment (oyershov) [inline src/containers/OpenOrders/index.tsx:109]: indentation😀

**Review by oyershov — APPROVED** [review]

**Pull request merged**